modules/sitemaps/sitemaps.php

```diff
@@ -1,792 +1,470 @@
- * Intended for mapping namespace and namespace URI values.
- * Passes array through jetpack_sitemap_ns for other functions to add their own namespaces.
- *
- * @module sitemaps
- *
- * @return array array of attribute value pairs passed through the jetpack_sitemap_ns filter
- */
-function jetpack_sitemap_namespaces() {
 	/**
-	 * Filter the attribute value pairs used for namespace and namespace URI mappings.
-	 *
-	 * @module sitemaps
-	 *
-	 * @since 3.9.0
-	 *
-	 * @param array $namespaces Associative array with namespaces and namespace URIs.
+	 * @since 4.6.0
+	 * @var int SITEMAP_INTERVAL Number of seconds between sitemap generations.
 	 */
-	return apply_filters( 'jetpack_sitemap_ns', array(
-		'xmlns:xsi'          => 'http://www.w3.org/2001/XMLSchema-instance',
-		'xsi:schemaLocation' => 'http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd',
-		'xmlns'              => 'http://www.sitemaps.org/schemas/sitemap/0.9',
-		// Mobile namespace from http://support.google.com/webmasters/bin/answer.py?hl=en&answer=34648
-		'xmlns:mobile'       => 'http://www.google.com/schemas/sitemap-mobile/1.0',
-		'xmlns:image'        => 'http://www.google.com/schemas/sitemap-image/1.1',
-	) );
-}
-
-/**
- * Start sitemap XML document, writing its heading and <urlset> tag with namespaces.
- *
- * @module sitemaps
- *
- * @param $charset string Charset for current XML document.
- *
- * @return string
- */
-function jetpack_sitemap_initstr( $charset ) {
-	global $wp_rewrite;
-	// URL to XSLT
-	if ( $wp_rewrite->using_index_permalinks() ) {
-		$xsl = home_url( '/index.php/sitemap.xsl' );
-	} else if ( $wp_rewrite->using_permalinks() ) {
-		$xsl = home_url( '/sitemap.xsl' );
-	} else {
-		$xsl = home_url( '/?jetpack-sitemap-xsl=true' );
-	}
-
-	$initstr = '<?xml version="1.0" encoding="' . $charset . '"?>' . "\n";
-	$initstr .= '<?xml-stylesheet type="text/xsl" href="' . esc_url( $xsl ) . '"?>' . "\n";
-	$initstr .= '<!-- generator="jetpack-' . JETPACK__VERSION . '" -->' . "\n";
-	$initstr .= '<urlset';
-	foreach ( jetpack_sitemap_namespaces() as $attribute => $value ) {
-		$initstr .= ' ' . esc_html( $attribute ) . '="' . esc_attr( $value ) . '"';
-	}
-	$initstr .= ' />';
-
-	return $initstr;
-}
-
-/**
- * Load XSLT for sitemap.
- *
- * @module sitemaps
- *
- * @param string $type XSLT to load.
- */
-function jetpack_load_xsl( $type = '' ) {
-
-	$transient_xsl = empty( $type ) ? 'jetpack_sitemap_xsl' : "jetpack_{$type}_sitemap_xsl";
-
-	$xsl = get_transient( $transient_xsl );
-
-	if ( $xsl ) {
-		header( 'Content-Type: ' . jetpack_sitemap_content_type(), true );
-		echo $xsl;
-		die();
-	}
-
-	// Populate $xsl. Use $type.
-	include_once JETPACK__PLUGIN_DIR . 'modules/sitemaps/sitemap-xsl.php';
-
-	if ( ! empty( $xsl ) ) {
-		set_transient( $transient_xsl, $xsl, DAY_IN_SECONDS );
-		echo $xsl;
-	}
-
-	die();
-}
+	const SITEMAP_INTERVAL = 60;
```

Comment:
Also re. scaling issues- I'm not very familiar with how wp_cron behaves, especially on slow hosts. If timing out is an issue, I tried to design this to be extensible so that sitemap generation can be paused and resumed. For example, instead of refreshing everything every 24 hours, it would refresh as much as it can before some timer runs out.

To be clear, this is *not* what the plugin does now, because I wasn't sure that it would be needed. But the design is such that adding that functionality should be mostly straightforward.